react/features/invite/components/security-dialog/PasswordSection.js

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable react/no-multi-comp */
+// @flow
+import React, { useRef } from 'react';
+
+import PasswordForm from './PasswordForm';
+import { translate } from '../../../base/i18n';
+import { copyText } from '../add-people-dialog';
+
+type Props = {
+
+    /**
+     * Whether or not the current user can modify the current password.
+     */
+    canEditPassword: boolean,
+
+    /**
+     * The JitsiConference for which to display a lock state and change the
+     * password.
+     */
+    conference: Object,
+
+    /**
+     * The value for how the conference is locked (or undefined if not locked)
+     * as defined by room-lock constants.
+     */
+    locked: string,
+
+    /**
+     * The current known password for the JitsiConference.
+     */
+    password: string,
+
+    /**
+     * Whether or not to show the password in editing mode.
+     */
+    passwordEditEnabled: boolean,
+
+    /**
+     * The number of digits to be used in the password.
+     */
+    passwordNumberOfDigits: ?number,
+
+    /**
+     * Action that sets the conference password.
+     */
+    setPassword: Function,
```

Comment:
General observation: It's not very good to pass these kind of functions too deep down in the react dom tree, which seems to be happening here. Can't these props be constructed in the component where they are being used? E.g. by using `mapStateToProps` or `mapDispatchToProps`?

This concern applies to most other props of this component too.

Reply:
I was trying to achieve a controlled component type of implementation, where the dialog is the parent. If you're adamant about this, I could change it, no problem